modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v8_9_3/CommerceOrderUpgradeProcess.java

```diff
@@ -0,0 +1,135 @@
+		CommerceOrderLocalService commerceOrderLocalService,
+		CommerceShipmentLocalService commerceShipmentLocalService,
+		CommerceOrderStatusRegistry commerceOrderStatusRegistry) {
+
+		_commerceOrderLocalService = commerceOrderLocalService;
+		_commerceShipmentLocalService = commerceShipmentLocalService;
+		_commerceOrderStatusRegistry = commerceOrderStatusRegistry;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_updateCommerceOrder();
+	}
+
+	private void _updateCommerceOrder() throws Exception {
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				"select commerceOrderId, orderStatus from CommerceOrder " +
+					"where orderStatus = ? or orderStatus = ? ")) {
+
+			preparedStatement.setInt(
+				1, CommerceOrderConstants.ORDER_STATUS_PARTIALLY_SHIPPED);
+			preparedStatement.setInt(
+				2, CommerceOrderConstants.ORDER_STATUS_SHIPPED);
+
+			try (ResultSet resultSet = preparedStatement.executeQuery()) {
+				while (resultSet.next()) {
+					int orderStatus = resultSet.getInt(2);
+
+					if (orderStatus ==
+							CommerceOrderConstants.
+								ORDER_STATUS_PARTIALLY_SHIPPED) {
+
+						_updateShippedCommerceOrderStatus(
+							_commerceOrderLocalService.getCommerceOrder(
+								resultSet.getLong(1)));
+					}
+					else if (orderStatus ==
+								CommerceOrderConstants.ORDER_STATUS_SHIPPED) {
+
+						_updateCompletedCommerceOrderStatus(
+							_commerceOrderLocalService.getCommerceOrder(
+								resultSet.getLong(1)));
+					}
+				}
+			}
+		}
+	}
+
+	private void _updateCompletedCommerceOrderStatus(
+			CommerceOrder commerceOrder)
+		throws Exception {
+
+		CommerceOrderStatus completedCommerceOrderStatus =
+			_commerceOrderStatusRegistry.getCommerceOrderStatus(
+				CompletedCommerceOrderStatusImpl.KEY);
+
+		int[] commerceShipmentStatuses =
+			_commerceShipmentLocalService.
+				getCommerceShipmentStatusesByCommerceOrderId(
+					commerceOrder.getCommerceOrderId());
+
+		if (completedCommerceOrderStatus.isTransitionCriteriaMet(
+				commerceOrder) &&
+			(commerceShipmentStatuses.length == 1) &&
+			(commerceShipmentStatuses[0] ==
+				CommerceShipmentConstants.SHIPMENT_STATUS_DELIVERED)) {
+
+			runSQL(
+				StringBundler.concat(
+					"update CommerceOrder set orderStatus = ",
+					CommerceOrderConstants.ORDER_STATUS_COMPLETED,
+					" where commerceOrderId = ",
+					commerceOrder.getCommerceOrderId()));
+		}
+	}
+
+	private void _updateShippedCommerceOrderStatus(CommerceOrder commerceOrder)
+		throws Exception {
+
+		CommerceOrderStatus shippedCommerceOrderStatus =
+			_commerceOrderStatusRegistry.getCommerceOrderStatus(
+				ShippedCommerceOrderStatusImpl.KEY);
+
+		if (shippedCommerceOrderStatus.isTransitionCriteriaMet(commerceOrder)) {
+			runSQL(
+				StringBundler.concat(
+					"update CommerceOrder set orderStatus = ",
+					CommerceOrderConstants.ORDER_STATUS_SHIPPED,
+					" where commerceOrderId = ",
+					commerceOrder.getCommerceOrderId()));
```

Comment:
@luca-pellizzon , I opted to just run a SQLQuery to update the orderStatuses because our orderModelListener will eventually trigger following:  [_sendOrderStatusMessage](https://github.com/liferay-commerce/liferay-portal/blob/85d463b5f4b24f7a28df8b331764c9692389e444/modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/order/engine/CommerceOrderEngineImpl.java#L550) which makes it impossible for us to use CommerceOrderEngineImpl as an update tool because our DTOs call *Service which always checks for permission. PermissionChecker is not available during the upgradeProcess.

Reply:
We could brute force it by just doing another modified check of `_updateCompletedCommerceOrderStatus` if `_updateShippedCommerceOrderStatus` passes, but I do not think that is the right approach because that would just add another technical debt to our root cause of CommerceOrderEngineImpl. Let me know what you think about this. If you are okay with it, then I can add another commit to do the brute force update.

Reply:
SQL query is the right way 👍

Reply:
@luca-pellizzon , is the current PR as is good? or did you want me to brute force the status to Complete?

Reply:
In the upgrade process, yes. I think we should leave everything as it should, so we set as complete the orders that have all the requirements.